web/packages/extension/src/common.ts

```diff
@@ -32,6 +32,35 @@ class CheckboxOption implements OptionElement<boolean> {
     }
 }
 
+class MaxExecutionDuration_NumberInputOption
+    implements OptionElement<Duration>
+{
+    constructor(
+        private readonly numberInput: HTMLInputElement,
+        readonly label: HTMLLabelElement
+    ) {}
+
+    get input() {
+        return this.numberInput;
+    }
+
+    get value() {
+        return Math.max(1, Math.round(parseInt(this.numberInput.value)));
+    }
+
+    set value(value: Duration) {
+        let newValue = 0;
```

Comment:
nit: `newValue` can be left uninitialized here, because all following code paths initialize it.